### ISSUE TITLE
Pre-publish panel calendar and timepicker are misaligned at certain breakpoints

### ISSUE THREAD

**Issue by sarahmonster** [WordPress/gutenberg#7863]
## Issue Overview

In the pre-publish panel, the calendar is left-aligned, whilst the timepicker below it is centre-aligned. This works at the majority of viewports where the sidebar is a controlled width overall, but falls apart at those viewports where the sidebar ends up being the full width of the screen: 

<img width="728" alt="screenshot 2018-07-09 20 23 47" src="https://user-images.githubusercontent.com/376315/42506977-75b7e390-843b-11e8-8639-acf9e32b60dd.png">

## Steps to Reproduce
1. Set your browser width between 320 and 778 pixels. (Approximately.)
2. Start a new post and add a title.
3. Press publish.
4. Click on the "Publish" panel to open the datepicker.
5. See misalignment!

This issue doesn't seem to happen elsewhere—the only other usage of the calendar component I could find was in the document settings sidebar, where the calendar appears as a popup instead:

<img width="730" alt="screenshot 2018-07-10 13 01 24" src="https://user-images.githubusercontent.com/376315/42508704-5e2799b8-8441-11e8-8463-6dc2028bd993.png">

## Possible Solution

Basically, this would be resolved either by setting a max-width on the timepicker, or left-aligning the timepicker. The former is more elegant and in line with the calendar component as used elsewhere, but does rely on the calendar retaining a set width.
